server/rooms.ts

```diff
@@ -1130,11 +1126,13 @@ export class GlobalRoomState {
 				if (!p1userid || !p2userid) continue;
 				if (!p1userid.startsWith(usernameFilter) && !p2userid.startsWith(usernameFilter)) continue;
 			}
-			if (skipCount && skipCount--) continue;
-
 			rooms.push(room);
 		}
 
+		if (rooms.length > 150 && !formatFilter && !eloFilter && !usernameFilter) {
+			rooms = rooms.slice(-150);
+		}
+
```

Comment:
```suggestion
```

The original cap of 150 was for performance reasons. This line doesn't actually impact performance.